src/content/posts/docker_volume_introduction.md

```diff
@@ -0,0 +1,84 @@
+
+So we need to find a way to **keep certain files after the container's death** and that can be **achieved with volumes**!.
+
+## How to create a volume?
+
+To create a volume, you need to add a specific argument to the `docker run` command.
+
+> You can also create volumes outside of containers with [`docker volume`](https://docs.docker.com/storage/volumes/#create-and-manage-volumes)
+
+There is two different arguments which are really similar in their use. Those are `--mount` and `--volume`, both can be used to create volumes (or bind mounts, see Disclaimer above).  
+However they slightly differ: if you enter a non existing file to mount, `--volume` will create it when `--mount` will return an error, also `--volume` needs fewer arguments. For these reason and because we focused on volumes in this post we'll describe the `--volume` argument. 
+
+The exact syntax is:
+```bash
+docker run <other args> --volume PATH_TO_HOST_FILE:PATH_TO_CONTAINER_FILE:<optional args>
+```
+
+> Note that you can use `-v` instead of `--volume`.
+
+> Note that you can mount a directory instead of a file with the same method.
+
+> A similar syntax exists for [`docker-compose.yml`](https://docs.docker.com/compose/compose-file/compose-file-v3/#volumes)
+
+Optional args must be separated by comas, they are not used very often but lets detail the most useful one quickly :
+- `ro` : if used, the mounted files/folders will be in read only mode inside the container
+
+A concrete example with the [code of our RootMe discord bot](https://github.com/iScsc/RootPythia/blob/2681ca26286ea5063371536e995a5e3cf39734a5/run.sh#L12):
+```bash
+    source ./.env.prod
+
+    docker run --rm --interactive --tty \
+	--detach \
+	--volume $(realpath -P ${LOG_FOLDER}):/opt/${NAME}/logs \
+	--env-file .env.prod \
+	--name ${NAME} \
+	${NAME}:latest
+```
+In this example `LOG_FOLDER` is an environment variable defined in the `.env.prod` file.  
+The code will write transparently to `/opt/${NAME}/logs/<log file>` but because this folder is shared with the host, logs will **be available in the host** and survive potential application crashes or updates.
+
+> **WARNING : Be careful that the bot/application running in the container must have the right permission to modify the file/directory**
+
+## Other important use cases of volumes
+
+* The same volume can be shared by multiple containers.  
+ It can be useful if you want to share a database with applications running on different containers or to centralized logs of multiple application in the same log file/folder.
+* In development you can use volumes to share code with the container (where it runs) while editing it in your IDE in the host.
```

Comment:
In this case, bind mounts are a better option
See https://docs.docker.com/storage/#good-use-cases-for-bind-mounts